### PR TITLE
Auth + Homepage

### PR DESCRIPTION
**Pull request by tejas-muthya37**
This PR contains the code for Authentication and Homepage of the app.

The code to be reviewed is present in src -> Screens -> Authenticate, src -> Screens -> Homepage and src -> Context

### CODE REVIEW

File: src/Screens/Authenticate/Authenticate.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import "./authenticate.css";
+import { useNavigate } from "react-router-dom";
+import { useRef } from "react";
+import { useToast } from "./../../Context/toast-context";
+import { useNavbar } from "./../../Context/navbar-context";
+import { useToken } from "./../../Context/token-context";
```
Comment (yaswanthmyneni) [inline src/Screens/Authenticate/Authenticate.js:6]: suggestion: create index.js in the context page and export all contexts from the index.js

File: src/Screens/Authenticate/Authenticate.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          <div className="secondary-cta-checkbox">
+            <input checked id="checked-checkbox" type="checkbox" />
+            <label for="checked-checkbox">{props.checkboxLabel}</label>
+          </div>
+          {props.cardTitle === "LOGIN" && (
+            <div>
+              <a href="/">Forgot Password</a>
+            </div>
+          )}
+        </div> */}
```
Comment (yaswanthmyneni) [inline src/Screens/Authenticate/Authenticate.js:130]: changes: remove unused code.

File: src/Screens/Authenticate/Authenticate.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+        body: JSON.stringify(payload),
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          console.log(data);
+          if (!data.errors) {
```
Comment (yaswanthmyneni) [inline src/Screens/Authenticate/Authenticate.js:39]: changes: remove console.log() while pushing to github

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes.

File: src/Context/token-context.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { createContext, useContext, useState } from "react";
+
+const TokenContext = createContext();
+
+const TokenProvider = ({ children }) => {
+  var encodedTokenLocal = localStorage.getItem("ENCODED_TOKEN_2");
```
Comment (Tejinder15) [inline src/Context/token-context.js:6]: avoid using var. either use let or const

File: src/Screens/Authenticate/Authenticate.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  const passwordRef = useRef(null);
+
+  const handleAuth = () => {
+    var payload = {
+      email: emailRef.current.value,
+      password: passwordRef.current.value,
+    };
+
+    if (props.cardTitle === "LOGIN") {
+      fetch("/api/auth/login", {
```
Comment (Tejinder15) [inline src/Screens/Authenticate/Authenticate.js:28]: fetch is fine but try using axios. it is more convinient and easier to work with. Also it makes code look short and clean.

File: src/Screens/Authenticate/Authenticate.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+          </div>
+          <div className="label-with-input">
+            <label id="password-label" htmlFor="password">
+              Password *
+            </label>
+            <input ref={passwordRef} type="password" id="password" />
+          </div>
+        </div>
+
+        {/* Will be implemented later */}
```
Comment (Tejinder15) [inline src/Screens/Authenticate/Authenticate.js:118]: you can remove comments if not needed.

File: src/Screens/Authenticate/Authenticate.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          </div>
+          {props.cardTitle === "LOGIN" && (
+            <div>
+              <a href="/">Forgot Password</a>
+            </div>
+          )}
+        </div> */}
+        <button onClick={handleAuth}>Next</button>
+        <p className="alternate-cta">
+          <a href="/">
```
Comment (Tejinder15) [inline src/Screens/Authenticate/Authenticate.js:133]: avoid using anchor tag as it will lead to reloading page. use <Link> tag in place of it

**Review by Tejinder15 — APPROVED** [review]
Other than the changes suggested the code looks fine.

**Pull request merged**